src/library/VTKExtensions/Rendering/vtkF3DInteractorEventRecorder.cxx

```diff
@@ -75,3 +78,113 @@ void vtkF3DInteractorEventRecorder::ProcessEvents(
     self->OutputStream->flush();
   }
 }
+
+#if VTK_VERSION_NUMBER >= VTK_VERSION_CHECK(9, 1, 20220510)
+//------------------------------------------------------------------------------
+void vtkF3DInteractorEventRecorder::ReadEvent(const std::string& line)
+{
+  // Read events and invoke them on the object in question
+  char event[256] = {}, keySym[256] = {};
+  int pos[2], ctrlKey, shiftKey, altKey, keyCode, repeatCount;
+  float tempf;
+
+  std::istringstream iss(line);
+
+  // Use classic locale, we don't want to parse float values with
+  // user-defined locale.
+  iss.imbue(std::locale::classic());
+
+  iss.width(256);
+  iss >> event;
+
+  // Quick skip comment
+  if (*event == '#')
+  {
+    // Parse the StreamVersion (not using >> since comment could be empty)
+    // Expecting: # StreamVersion x.y
+
+    if (strlen(line.c_str()) > 16 && !strncmp(line.c_str(), "# StreamVersion ", 16))
+    {
+      int res = sscanf(line.c_str() + 16, "%f", &tempf);
+      if (res && res != EOF)
+      {
+        this->CurrentStreamVersion = tempf;
+      }
+    }
+  }
+  else
+  {
+    if (this->CurrentStreamVersion == 0)
+    {
+      vtkWarningMacro("StreamVersion has not been read, parsing may be incorrect");
+    }
+
+    unsigned long ievent = vtkCommand::GetEventIdFromString(event);
+    if (ievent != vtkCommand::NoEvent)
+    {
+      iss >> pos[0];
+      iss >> pos[1];
+      if (this->CurrentStreamVersion >= 1.1)
+      {
+        int m;
+        iss >> m;
+        shiftKey = (m & ModifierKey::ShiftKey) ? 1 : 0;
+        ctrlKey = (m & ModifierKey::ControlKey) ? 1 : 0;
+        altKey = (m & ModifierKey::AltKey) ? 1 : 0;
+      }
+      else
+      {
+        iss >> ctrlKey;
+        iss >> shiftKey;
+        altKey = 0;
+      }
+      iss >> keyCode;
+      iss >> repeatCount;
+      iss >> keySym;
+
+      vtkSmartPointer<vtkStringArray> strArr;
+      vtkIdType nStr;
+      iss >> nStr;
+      if (nStr > 0)
+      {
+        strArr = vtkSmartPointer<vtkStringArray>::New();
+        for (int i = 0; i < nStr; i++)
+        {
+          std::string str;
+          iss >> str;
+          strArr->InsertNextValue(str);
+        }
+      }
+
+      this->Interactor->SetEventPosition(pos);
+      this->Interactor->SetControlKey(ctrlKey);
+      this->Interactor->SetShiftKey(shiftKey);
+      this->Interactor->SetAltKey(altKey);
+      this->Interactor->SetKeyCode(static_cast<char>(keyCode));
+      this->Interactor->SetRepeatCount(repeatCount);
+      this->Interactor->SetKeySym(keySym);
+
+      this->Interactor->InvokeEvent(ievent, strArr);
```

Comment:
@Meakk this is a copied from VTK with the addition of the strings at the end. While this is fine this is not perfect has our logs are not respective the VTK interactive log standard, if there is such a things, which we may be impacted if it ever evolved in the future. I could add a more generic approach in VTK, what do you think ?

Reply:
At this point, I think we should have our own serialization based on json for example.

Reply:
I dont think so. Do you plan to rewrite the whole logic in json ?

I think that what we need can be put in VTK and improving VTK gives us free maintenance on the code we put in VTK.

Reply:
If we improve the VTK version, it will still not work with previous VTK versions, right?

Reply:
it will, thanks tonthe "StreamVersion" logic

Reply:
I'll move forward with a VTK side implementation. I think it make sense.